pkg/client/unversioned/jobs.go

```diff
@@ -85,7 +89,7 @@ func (c *jobs) Delete(name string, options *api.DeleteOptions) (err error) {
 		return c.r.Delete().Namespace(c.ns).Resource("jobs").Name(name).Do().Error()
 	}
 
-	body, err := api.Scheme.EncodeToVersion(options, c.r.APIVersion())
+	body, err := api.Scheme.EncodeToVersion(options, latest.GroupOrDie("").GroupVersion)
```

Comment:
I think the horizontalpodautoscaler code will also fail if that line is executed. I think Eric's fix is a temporary one. In the future, apart from the ExperimentalClient, struct like ehorizontalpodautoscaler shall contain a RESTClient as well.